src/alloc.rs

```diff
@@ -40,6 +42,8 @@ unsafe impl GlobalAlloc for GlobalAllocator {
 
 unsafe impl Alloc for GcAllocator {
     unsafe fn alloc(&mut self, layout: Layout) -> Result<NonNull<u8>, AllocErr> {
+        COLLECTOR.lock().poll();
```

Comment:
I sort-of assumed we only want to `lock` once? Admittedly, I can't remember if unlocking is automatic or not...

Reply:
Yeah I agree it's not ideal to lock the collector to poll it, and then immediately lock it again afterwards to bump `allocations`. Let me rewrite this with a temporary so we acquire the lock only once.

Reply:
Fixed: ce0fa64d22f53af694df517f2b46b02c94a742c0